Calling/App.xaml.cs

```diff
@@ -86,5 +147,65 @@ private void OnSuspending(object sender, SuspendingEventArgs e)
             //TODO: Save application state and stop any background activity
             deferral.Complete();
         }
+
+        private async Task InitNotificationsAsync()
+        {
+            if (AZURE_PNH_HUB_NAME != "<AZURE_PNH_HUB_NAME>" && AZURE_PNH_HUB_CONNECTION_STRING != "<AZURE_PNH_HUB_CONNECTION_STRING>")
+            {
+                var channel = await PushNotificationChannelManager.CreatePushNotificationChannelForApplicationAsync();
+                channel.PushNotificationReceived += Channel_PushNotificationReceived;
+
+                var hub = new NotificationHub(AZURE_PNH_HUB_NAME, AZURE_PNH_HUB_CONNECTION_STRING);
+                var result = await hub.RegisterNativeAsync(channel.Uri);
+
+                if (result.ChannelUri != null)
+                {
+                    PNHChannelUri = result.ChannelUri.ToString();
+                }
+                else
+                {
+                    Debug.WriteLine("Cannot register WNS channel");
+                }
+            }
+        }
+
+        private async void Channel_PushNotificationReceived(PushNotificationChannel sender, PushNotificationReceivedEventArgs args)
+        {
+            args.Cancel = true;
+
+            switch (args.NotificationType)
+            {
+                case PushNotificationType.Toast:
+                case PushNotificationType.Tile:
+                case PushNotificationType.TileFlyout:
+                case PushNotificationType.Badge:
+                    break;
+                case PushNotificationType.Raw:
+                    var frame = (Frame)Window.Current.Content;
+                    if (frame.Content is MainPage)
+                    {
+                        var mainPage = (MainPage)frame.Content;
```

Comment:
Maybe a soft-cast:

var mainPage = frame.Content as MainPage;

And you can null check on mainPage without having to throw.

Reply:
Applied soft cast